### PR TITLE
Feat/next lesson card LESQ-409

### PR DESCRIPTION
**Pull request by k-huggs**
## Description

- Creates NextLessonCard component, test file and story

## Issue(s)

Fixes #LESQ-409

## How to test

1. Go to https://deploy-preview-2075--owa-storybook.netlify.thenational.academy/?path=/docs/download-components-nextlessoncard--docs

## Screenshots

How it should now look:
Screenshot 2023-11-06 at 13 37 32

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

**Review by jack-skerrett-oak — COMMENTED** [review]
You haven't asked for a review yet but I saw the PR :D 

Just noticed at mobile widths it cuts off the "Download resources" copy. Didn't know if that was a storybook issue, or would annoy us on the front end eventually?
Screenshot 2023-11-06 at 14 14 12

**Comment by k-huggs** [comment]
> You haven't asked for a review yet but I saw the PR :D
> 
> Just noticed at mobile widths it cuts off the "Download resources" copy. Didn't know if that was a storybook issue, or would annoy us on the front end eventually? Screenshot 2023-11-06 at 14 14 12

This is a width of the card and size of the button issue, still trying to figure out if I can solve it

File: src/components/DownloadComponents/NextLessonCard/NextLessonCard.test.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    const { getByTestId } = renderWithTheme(
+      <NextLessonCard
+        lessonTitle="test-lesson"
+        programmeSlug="test-programme"
+        unitSlug="test-unit"
+        lessonSlug="test-slug"
+      />,
+    );
+
+    const seeLessonLink = getByTestId("see-lesson-link");
```
Comment (njblore) [inline src/components/DownloadComponents/NextLessonCard/NextLessonCard.test.tsx:30]: can we `getByRole` instead of using testId?

Comment (k-huggs) [reply src/components/DownloadComponents/NextLessonCard/NextLessonCard.test.tsx]: Amended ✅

**Review by jack-skerrett-oak — APPROVED** [review]

**Review by njblore — APPROVED** [review]

**Review by Monika-Koziol — APPROVED** [review]
Looking good.

**Pull request merged**